<p>
    Class methods that don't access instance data can and should be <code>static</code> to prevent any misunderstanding
    about the contract of the method.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
public class Utilities
{
  public int Sum(int a, int b)  // Noncompliant
  {
    return a + b;
  }
}
</pre>

<h2>Compliant Solution</h2>
<pre>
public class Utilities
{
  public static int Sum(int a, int b)  // Compliant
  {
    return a + b;
  }
}
</pre>
