<p>
When exceptions occur, it is usually a bad idea to simply ignore them. Instead, it is better to handle them properly, or at least to log them.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
string text = "";
try {
  text = File.ReadAllText(fileName);
} catch (Exception exc) {  // Noncompliant
}
</pre>

<h2>Compliant Solution</h2>
<pre>
string text = "";
try {
  text = File.ReadAllText(fileName);
} catch (Exception exc) {
  logger.Log(exc);
}
</pre>