<p>The use of <code>==</code> to compare to objects is expected to do a reference comparison. That is, it is expected to return <code>true</code> if
and only if they are the same object instance. Overloading the operator to do anything else will inevitably lead to the introduction bugs by callers.
On the other hand, overloading it to do exactly that is pointless; that's what <code>==</code> does by default.</p>
<h2>Noncompliant Code Example</h2>
<pre>
public static bool operator== (MyType x, MyType y) // Noncompliant
{
</pre>
<h2>Exceptions</h2>
<ul>
  <li> If the class overloads the <code>operator +</code> or the <code>operator -</code> then this rule won't raise. </li>
  <li> If the class implements the <code>IComparable&lt;T&gt;</code> interface then the class most probably behaves as a value-type object so the rule
  won't raise. </li>
  <li> If the class implements the <code>IEquatable&lt;T&gt;</code> interface then the class most probably behaves as a value-type object so the rule
  won't raise. </li>
</ul>

